### ISSUE TITLE
Ability to apply transformation to axis labels

### ISSUE THREAD

**Issue by l4cr0ss** [jamesleesaunders/d3-x3dom#59]
Pardon if StackOverflow is a better place to put this. If it is I will close and open question there.

I want to be able to apply a matrix transformation to the labels on the axes so that they are easier to read from the default viewport. Is there a straightforward way to do this? If not where is the best place to add some code to do so? Happy to make the pull request if you can help me locate the right place to make the change.

**Comment by jamesleesaunders** [comment]
Hi @l4cr0ss thanks for your interest in d3-x3dom. I am happy to help where I can, and even more happy for your offer of a pull request should we need it.

At the moment the axis labels are attached to a billboard element which rotates to face the viewport. If you have another requirement I would love to work with you on a pull request.
The files you would need to look into are:
  /src/component/axis.js
  /src/component/axisThreePlane.js

I don’t think you will need to touch the ThreePlane file as this just combines three axis components. The bit of axis.js you may need to hone into is:

```
if (tickFormat !== "") {
  let text = tick.select("billboard");
  let newText = tickEnter.append("transform");
  newText
    .attr("translation", tickDirectionVector.map((d) => (-d * tickPadding)))
    .append("billboard")
    .attr("axisofrotation", "0 0 0")
    .append("shape")
    .call(makeSolid, "black")
    .append("text")
    .attr("string", tickFormat)
    .append("fontstyle")
    .attr("size", 1.3)
    .attr("family", "SANS")
    .attr("style", "BOLD")
    .attr("justify", "MIDDLE");
  text = text.merge(newText);
```

You may also be interested in this example on Observable:
https://observablehq.com/@jamesleesaunders/d3-x3d-components-part-1-axis

All the best,
Jim

**Comment by l4cr0ss** [comment]
Thank you for your comment - I've been digging around since I first posted and you've confirmed that I was digging in the right place.

I'm in the process of using your library to build a 3 dimensional bar chart. What's provided out of the box by your library is 90% of what I need, except that the labels are very hard to read.

I just submitted a pull request - one which is far from being ready to pull - but I am hopeful that it will give you an idea of what I'm trying to do.

This is my first time using Github (most of my work ends up in an on-premise Gitlab instance) so I'm sorry I haven't linked up the pull request to this issue. I am learning as I go.
